app-testing/src/pages/protocol_landing_v5dot1.py

```diff
@@ -0,0 +1,505 @@
+
+    def get_protocol_library_link(self) -> Optional[WebElement]:
+        """Get the Protocol Library Link on Protocol Page."""
+        header: Element = Element(
+            (By.ID, "EmptyStateLinks_protocolLibraryButton"),
+            "the Protocol Library Link on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_protocol_designer_link(self) -> Optional[WebElement]:
+        """Get the Protocol Designer Link on Protocol Page."""
+        header: Element = Element(
+            (By.ID, "EmptyStateLinks_protocolDesignerButton"),
+            "the Protocol Designer Link on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_python_api_link(self) -> Optional[WebElement]:
+        """Get the python api Link on Protocol Page."""
+        header: Element = Element(
+            (By.ID, "EmptyStateLinks_apiDocsButton"),
+            "the python api Link on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_show_in_folder(self) -> Optional[WebElement]:
+        """Get show in folder in overflow button on Protocol Page."""
+        header: Element = Element(
+            (By.XPATH, "//button[@data-testid='ProtocolOverflowMenu_showInFolder']"),
+            "show in folder in overflow button on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_run_protocol(self) -> Optional[WebElement]:
+        """Get run in overflow button on Protocol Page."""
+        header: Element = Element(
+            (By.XPATH, "//button[@data-testid='ProtocolOverflowMenu_run']"),
+            "run in overflow button on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_delete_protocol(self) -> Optional[WebElement]:
+        """Get delete protocol in overflow button on Protocol Page."""
+        header: Element = Element(
+            (By.XPATH, "//button[@data-testid='ProtocolOverflowMenu_deleteProtocol']"),
+            "delete protocol in overflow button on Protocol Page.",
+        )
+        return self.base.present_wrapper(header, 2)
+
+    def get_import_button_protocol_landing(self) -> Optional[WebElement]:
+        """Get the import button on Protocol Landing Page."""
+        header: Element = Element(
+            (By.XPATH, '//button[text()="Import"]'),
+            "the import button on Protocol Landing Page.",
+        )
+        return self.base.present_wrapper(header, 4)
+
+    def get_deckMap_protocol_landing(self) -> Optional[WebElement]:
+        """Get the deckmap on Protocol Landing Page."""
+        header: Element = Element(
+            (
+                By.XPATH,
+                '//div[@data-testid="ProtocolCard_deckLayout_script_pur_sample_1"]',
+            ),
+            "the deckmap on Protocol Landing Page.",
+        )
+        return self.base.present_wrapper(header, 7)
```

Comment:
Is it possible in all the locators that include the protocolName `script_pur_sample_1` to intake this name as a parameter so this same locator function may be used for other protocols?
```suggestion
    def get_deckMap_protocol_landing(self, protocol_name: str) -> Optional[WebElement]:
        """Get the deckmap on Protocol Landing Page."""
        header: Element = Element(
            (
                By.XPATH,
                f'//div[@data-testid="ProtocolCard_deckLayout_{protocol_name}"]',
            ),
            "the deckmap on Protocol Landing Page.",
        )
        return self.base.present_wrapper(header, 7)
```

Reply:
Done